csharp/src/Ice/InputStream.cs

```diff
@@ -1015,6 +1049,8 @@ private int ReadAndCheckSeqSize(int minElementSize)
             return sz;
         }
 
+        private int ReadFixedLengthSize() => OldEncoding ? ReadInt() : ReadSize();
```

Comment:
same here this method already check the encoding, and ReadSize will do it again

Reply:
Fixed.